src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/diagnostics/BuildMemoryReport.java

```diff
@@ -0,0 +1,196 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sonyericsson.hudson.plugins.gerrit.trigger.diagnostics;
+
+import com.sonyericsson.hudson.plugins.gerrit.trigger.Messages;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.gerritnotifier.model.BuildMemory;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.ChangeBasedEvent;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.GerritTriggeredEvent;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.RefUpdated;
+import hudson.model.ModelObject;
+
+import javax.annotation.Nonnull;
+import java.text.SimpleDateFormat;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+/**
+ * Contains a snapshot clone of a {@link BuildMemory}.
+ *
+ * @author Robert Sandell &lt;rsandell@cloudbees.com&gt;.
+ * @see BuildMemory#report()
+ * @see Diagnostics
+ */
+public class BuildMemoryReport implements Map<GerritTriggeredEvent, List<BuildMemory.MemoryImprint.Entry>>, ModelObject {
+
+    private final Map<GerritTriggeredEvent, List<BuildMemory.MemoryImprint.Entry>> internal;
+
+    /**
+     * The format used to display timestamps.
+     * <p/>
+     * A variant of ISO 8601 with the 'T' replaced by a space for simpler ocular parsing.
+     */
+    public static final SimpleDateFormat TS_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ssZ");
+
+
+    /**
+     * Default Constructor.
+     */
+    public BuildMemoryReport() {
+        internal = new TreeMap<GerritTriggeredEvent, List<BuildMemory.MemoryImprint.Entry>>(
+                new Comparator<GerritTriggeredEvent>() {
+            @Override
+            public int compare(GerritTriggeredEvent a, GerritTriggeredEvent b) {
+                int to = a.getEventCreatedOn().compareTo(b.getEventCreatedOn()) * -1;
+                if (to == 0) {
+                    return Integer.compare(a.hashCode(), b.hashCode()) * -1;
+                }
+                return to;
+            }
+        });
+    }
+
+    /**
+     * Gets a sorted list of the contents from {@link #entrySet()}.
+     * The sorting is based on the inverse comparison of {@link GerritTriggeredEvent#getEventCreatedOn()}.
+     *
+     * @return a sorted list of this report's entries.
+     */
+    public List<Map.Entry<GerritTriggeredEvent, List<BuildMemory.MemoryImprint.Entry>>> getSortedEntrySet() {
```

Comment:
Not really, this is one of the more straight forward ways there are. But I guess you mean "better looking with less generics"? Then yes, but those are imho less straight forward ;)